source/grid/tria.cc

```diff
@@ -2068,11 +2068,22 @@ namespace internal
             // need to allow that
             if (boundary_line->boundary_id != numbers::internal_face_boundary_id)
               {
-                AssertThrow (! (line->boundary_id() == numbers::internal_face_boundary_id),
-                             ExcInteriorLineCantBeBoundary(line->vertex_index(0),
-                                                           line->vertex_index(1),
-                                                           boundary_line->boundary_id));
-                line->set_boundary_id (boundary_line->boundary_id);
+                if (line->boundary_id() == numbers::internal_face_boundary_id)
+                  {
+                    if (boundary_line->manifold_id == numbers::flat_manifold_id)
+                      {
+                        AssertThrow (! (line->boundary_id() == numbers::internal_face_boundary_id),
```

Comment:
Mmm, I have done this 2 weeks ago and Luca forced me to make the pull request now :)
I can't exactly recall what the problem was, but I'm quite sure it wouldn't work just changing the grid_in.h and grid_in.cc files. I'll make my checks and solve the problem. I'll also try to fix the other minor things and add a test.